Remove mission search radius from most missions

RenechCDDA opened the pull request:
#### Summary
None

#### Purpose of change
"Failed to assign mission target <name_of_OMT>"

We let our searches purposefully fail due to putting in a too-low number in json. Why?

There is absolutely no reason to let these searches fail on purpose. But for some reason, instead of just searching as far as reasonably possible, we let them fail. Not only that, but we leave it up to the JSON to determine just how often we want to purposefully fail, with [wildly varying results](https://github.com/CleverRaven/Cataclysm-DDA/pull/73925#pullrequestreview-2065078860). Why?!

Most of them are just cargo culted, and didn't understand what they were doing. Well let's lock it down.

#### Describe the solution
Guard against people accidentally setting a lower search range by enforcing `random`. If it's not random, it'll always search closest first and then you never want to stop the search early.

Upgrade the default search range from OMAPX (180 OMTs) to OMAPX * 14 (2520 OMTs). Upgrade the error to mention how far we searched, too. So that errors in search radius are apparent in debug messages, and don't have to be sussed out.

I audited every single search radius and kept search ranges for anything that previously had `random` search and seemed vaguely appropriate.

Conversely, I *removed* random (and did not return manual search range) from a few cases where it was clearly inappropriate. Anything searching for a GLOBAL_UNIQUE, for example, has no business searching an extra-limited range.

The end result is that all of our missions looking for singular important locations (i.e. all the ones which fail constantly) should be much more reliable.

#### Describe alternatives you've considered
Enforced 5000 tile search range

#### Testing
Loads. Properly errors if you try to lower the search range for no reason.

![image](https://github.com/user-attachments/assets/d7c43af4-9217-4d93-9de4-5cdea713d799)

#### Additional context

RenechCDDA committed:
Remove mission search radius from most missions

RenechCDDA commented on src/mission_util.cpp:
```diff
@@ -411,7 +411,7 @@ mission_target_params mission_util::parse_mission_om_target( const JsonObject &j
         p.random = jo.get_bool( "random" );
     }
-    p.search_range  = get_dbl_or_var( jo, "search_range", false, OMAPX );
+    p.search_range  = dbl_or_var( 2500.0 );
```
This is crap, but was the most straightforward way to plug in our new value without needlessly churning lots of C++.

moxian replied:
`search_range` is read in three places in the entire codebase, surely it should not be *too bad* replacing those three, with a [named] constant?

But sure, if we're trying to change as little c++ as possible, please at least add a comment that this is a misfeature and not some kind of big-brained plan.
```suggestion
    // Legacy code. Ideally this should not be a member field at all, since we hardcode it anyway.
    p.search_range  = dbl_or_var( 2500.0 );
```

RenechCDDA replied:
No longer relevant

John-Candlebury commented:
I think expanding the search radius is fine when it comes to missions trying to find unique placement specials.

However there's incredibly many reasons for why you'd want something to forcibly spawn within <X overmap tiles of the player, (mission and non mission related) and the mission spawning code is AFAIK the only way in which this can be achieved, so completely getting rid of the functionality is not a great idea.

For example I'm not sure how this will trully affect the spawning of Aftershock shuttles, if the player has explored enough fields around them, could the shuttle be placed so far away that they wont be able to reach it before it leaves?  I wouldnt want that to ever happen.

RenechCDDA commented:
> I think expanding the search radius is fine when it comes to missions trying to find unique placement specials.
> 
> However there's incredibly many reasons for why you'd want something to forcibly spawn within <X overmap tiles of the player, (mission and non mission related) and the mission spawning code is AFAIK the only way in which this can be achieved, so completely getting rid of the functionality is not a great idea.
> 
> For example I'm not sure how this will trully affect the spawning of Aftershock shuttles, if the player has explored enough fields around them, could the shuttle be placed so far away that they wont be able to reach it before it leaves? I wouldnt want that to ever happen.

get_target_position() (lambda) always tries to find **or** create on your overmap first. It only proceeds to other overmaps if it fails in your local overmap. The shuttle eoc looks like it can just be placed down in any field regardless of existing submaps.

TBH if we find issues with this it's a lot easier to re-add the functionality for the few cases that (might?) need it, than to allow it for every mission. Missions should definitely not be using this unless they explicitly need it, and (no offense intended) I am not convinced any missions have a need for it.

John-Candlebury commented:
But you still gave no valid reason to remove it? Like I see its being extensively misused, but you are completely getting rid of our ability  to say: "Game spawn this far enough but not too far away"

Sure you cannot think any reasons for why youd want to use the option in vanilla but that not true off in mods. Just familiarizing myself with the changes here, the sky island mod setting the exildistance between two values seems extremely deliberate and a valid use case to me, especially since it changes depending on choices the platyer makes?

Never mind that theres other valid reasons that are not implemented already: Example you use hellgate spell and the helgate spawns between 3 and 5 overmaps away from you (because you dont want to do the effects of spawning the thing live in the reality bubble but you dont want it too far away), same logic applies to for example Aftershock using a SAM site to down a drone/plane/spaceship.

RenechCDDA committed:
Re-add search ranges to some missions that actually were using the random range function
-Remove random from those that clearly should never have had it applied (and don't return their search range either)

RenechCDDA commented:
Okay after some discussion with Candlebury: They're completely right. That WAS doing something useful. Oops.

So I went through every single one of them, again, and re-added the search ranges to anything that previously had `random` search and seemed vaguely appropriate.

Conversely, I *removed* random (and did not return manual search range) from a few cases where it was clearly inappropriate. Anything searching for a GLOBAL_UNIQUE, for example, has no business searching a limited range. Dinomod had one case of searching only one overmap for a 4x4 (!!!) surface lab. That was incredibly prone to failure, so I just nixed that to make sure it could be found.

Other than that, very cut and dry. But they're in a separate commit so mod authors/reviewers can look them over to be sure.

Procyonae commented:
I don't hate this as a temporary measure but this needs a more nuanced permanent solution, I've thought quite a bit about ideal solutions but still haven't decided on anything, you can find some of my previous thoughts here https://github.com/CleverRaven/Cataclysm-DDA/pull/76684#issuecomment-2464520859

RenechCDDA committed:
Update data/mods/Magiclysm/items/currency.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Maleclypse committed:
Update forge_of_wonders.json
